includes/amp-helper-functions.php

```diff
@@ -1983,3 +1947,39 @@ function amp_generate_script_hash( $script ) {
 	);
 	return 'sha384-' . $hash;
 }
+
+/**
+ * Get the AMP version of a URL.
+ *
+ * @since 2.1
+ * @todo Rename this to clear up confusion with amp_get_current_url().
+ *
+ * @param string $url URL.
+ * @return string AMP URL.
+ */
+function amp_get_url( $url ) {
+	return add_query_arg( amp_get_slug(), '1', $url );
+}
+
+/**
+ * Determine whether the current request has the AMP query parameter set.
+ *
+ * @since 2.1
+ * @todo Rename this to be more agnostic to what the URL contains.
+ *
+ * @return bool True if the AMP query parameter is set with the required value, false if not.
+ * @global WP_Query $wp_query
+ */
+function amp_has_query_var() {
```

Comment:
Given that in the future a paired AMP URL will be able to be indicated arbitrarily, such as via a subdomain, I think we can anticipate that by making `amp_has_query_var()` more abstract. 

We have `amp_is_request()` to indicate that the request is being made for an AMP page. What we need is a method that says that the request was made for a paired AMP URL. What about `amp_is_paired_request()`? I'm not entirely satisfied with that since it implies that it entails `amp_is_request()`, being more specific. It's actually just giving an indication that the current URL is for paired AMP.

Maybe `amp_is_paired_url()`? We could pass it a `$url` argument as well, and it could tell you if _that_ URL is paired rather than the current request to WP. This would be useful in other contexts, like in the AMP-to-AMP Linking sanitizer perhaps.

Reply:
Thinking about this some more… what about `amp_is_paired_endpoint`? Using “endpoint” terminology would align with what we already have in `amp_remove_endpoint()` (which we could technically rename to `amp_remove_paired_endpoint()`).

While in a WordPress context, an “endpoint” refers to the _end_ of the URL, in reality an API endpoint doesn't necessarily refer to any physical part of the URL. Normally the endpoint will still be at the end (`?amp=1`) but sites can customize the paired AMP endpoint URLs to look however they want.

Reply:
I suppose then we'd rename `amp_get_paired_url( $url )` to `amp_get_paired_endpoint( $url )`.

Reply:
We basically need CRUD functions for manipulating paired AMP URLs:

* Read: `has`
* Create/Update: `get`

Reply:
So:

* Read: `amp_is_paired_endpoint( $url )`
* Create/Update: `amp_get_paired_endpoint( $url )`
* Delete: `amp_remove_paired_endpoint( $url )` <em>(formerly `amp_remove_endpoint()`)</em>

Thoughts?

Reply:
> Using “endpoint” terminology would align with what we already have in amp_remove_endpoint() (which we could technically rename to amp_remove_paired_endpoint()).

That reasoning makes sense. I'm also in support of renaming the functions to represent the CRUD methods of manipulating the paired URL :+1:.

Sidenote: That means `amp_remove_endpoint()` would have to be deprecated and annotated accordingly since its documented.

Reply:
Here, check this out: d5610f2